options/options_helper.cc

```diff
@@ -1438,6 +1329,61 @@ bool OptionTypeInfo::MatchesOption(const ConfigOptions& config_options,
   return false;
 }
 
+bool OptionTypeInfo::MatchesStruct(
+    const ConfigOptions& config_options, const std::string& struct_name,
+    const std::unordered_map<std::string, OptionTypeInfo>* struct_map,
+    const std::string& opt_name, const char* this_offset,
+    const char* that_offset, std::string* mismatch) {
```

Comment:
Would be better to use this_addr and that_addr instead of this_offset and that_offset.